### PR TITLE
ascanrules: Use Log4j 2.x

### PR DESCRIPTION
**Pull request by kingthorin**
- Update build file, CHANGELOGS, and change code to use updated logging infrastructure.
- Remove use of 2.10 snapshot in build files where it is no longer necessary.

Part of: zaproxy/zaproxy#6376

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
Looks like I broke the SQLi scan rule.  I suspect I simply buggered up the braces somewhere. Will dig into it tomorrow.

**Comment by kingthorin** [comment]
Fixed

File: addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/RemoteFileIncludeScanRule.java [hunk 1]
```diff
@@ -255,17 +247,12 @@ public void scan(HttpMessage msg, String param, String value) {
...
-                                + getBaseMsg().getRequestHeader().getMethod()
-                                + "] ["
-                                + getBaseMsg().getRequestHeader().getURI()
-                                + "], parameter ["
-                                + param
-                                + "] for Remote File Include. "
-                                + e.getMessage());
-            }
+            log.debug(
+                    "Error checking [{}] [{}], parameter [{}] for Remote File Include. ",
```
Comment (thc202) [inline addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/RemoteFileIncludeScanRule.java:251]: Missing `{}` at the end.

**Comment by thc202** [comment]
There's one update missing in `CommandInjectionScanRule` line 596.

**Comment by kingthorin** [comment]
Thanks, done.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thanks!

**Review by psiinon — APPROVED** [review]

**Comment by thc202** [comment]
@kingthorin could you rebase? (For Travis CI check.)

**Comment by kingthorin** [comment]
Sure, in-bound 😀

**Pull request merged**